module/zfs/zfs_vnops.c

```diff
@@ -1188,7 +1188,7 @@ zfs_rewrite(znode_t *zp, uint64_t off, uint64_t len, uint64_t flags,
 		error = dmu_buf_hold_array_by_dnode(dn, off, n, TRUE, FTAG,
 		    &numbufs, &dbp, DMU_READ_PREFETCH | DMU_UNCACHEDIO);
```

Comment:
It hardly matters, but I think you could move this up a few lines to before `dmu_tx_assign()` to avoid the need to abort it on error?

Reply:
We could, but I couldn't decide whether it is better to extend the TXG waiting for the hold, or extend the hold waiting for TXG.

Reply:
Yeah, I've got no opinion really, just spotted while walking past.